proposals/002_Token_Distribution_Proposal.md

```diff
@@ -0,0 +1,277 @@
+# [PROPOSAL-002] Guru Network Token Distribution Between Core Contributors
+
+## Introduction
+
+This proposal outlines an algorithm for distributing tokens among DAO members based on their contributions and the importance of their roles. The goal is to ensure a fair and transparent method for rewarding members based on their efforts and impact.
+
+## Overview
+
+The algorithm considers both individual contributions and role importance, using predefined weights to calculate each member's share of the total tokens. This method ensures that both the quantity and quality of work are recognized, as well as the varying significance of different roles within the organization.
+
+## Methodology
+
+### Weights for Roles and Metrics
+
+To achieve a balanced distribution, we will assign specific weights to different roles and metrics. These weights can be adjusted based on the organization's priorities and goals.
+
+#### Role Weights
+
+| Role       | Weight |
+|------------|--------|
+| Developer  | 1.0    |
+| Marketer   | 0.8    |
+| Designer   | 0.9    |
+| DevOps     | 1.1    |
+
+#### Metric Weights
+
+| Metric            | Weight |
+|-------------------|--------|
+| Tasks Completed   | 0.3    |
+| Time Efficiency   | 0.2    |
+| Task Complexity   | 0.2    |
+| Quality of Work   | 0.3    |
+
+### Evaluation of Individual Contributions
+
+Each member's contribution will be evaluated based on the following metrics:
+
+- **Tasks Completed:** Number of tasks completed during the period.
```

Comment:
Perhaps we should measure not just tasks, but also the features and system components designed, ensuring they correspond with our main mission (guru_framework and guru_network)?